### PR TITLE
Minor fixes for stackgraph generation

### PR DESCRIPTION
**Pull request by bknueven**
This PR makes the following enhancements to stackgraph generation:
1. Fills out fuel codes dictionary
1. Reserve shortfall now "covers" required reserve
1. Implicit reserves now calculated correctly when reserves are included

### CODE REVIEW

**Comment by bknueven** [comment]
@rconcep mind taking a look at this?

File: egret/viz/generate_graphs.py [hunk 1]
```diff
@@ -390,9 +396,9 @@ def _plot_generation_stack_components():
             reserves_by_hour += reserves_available
     
     if reserve_requirements_by_hour is not None:
-        implicit_reserves_by_hour = [max(0.0, reserves_by_hour[ix] - reserves_by_hour[ix]) for ix in range(len(reserve_requirements_by_hour))]
+        implicit_reserves_by_hour = np.maximum(0.0, reserves_by_hour - reserve_requirements_by_hour)
```
Comment (rconcep) [inline egret/viz/generate_graphs.py]: I think `reserve_requirements_by_hour` needs the `attribute_to_array` treatment. (See TravisCI build error (subtraction between float and dictionary))

File: egret/viz/generate_graphs.py [hunk 2]
```diff
@@ -352,22 +354,21 @@ def _plot_generation_stack_components():
     ax.step(demand_indices, demand_by_hour, linewidth=3, color='#000000', where='post')
     
     # Add reserve requirements, if applicable.
-    reserve_requirements_by_hour = egret_model_data.data['system'].get('reserve_requirement')
+    reserve_requirements_array = attribute_to_array(egret_model_data.data['system'].get('reserve_requirement'))
     if reserve_requirements_by_hour is not None:
```
Comment (rconcep) [inline egret/viz/generate_graphs.py]: I think this needs to be changed to `reserve_requirements_array` to fix the NameError

File: egret/viz/generate_graphs.py [hunk 3]
```diff
@@ -396,7 +397,7 @@ def _plot_generation_stack_components():
             reserves_by_hour += reserves_available
     
     if reserve_requirements_by_hour is not None:
```
Comment (rconcep) [inline egret/viz/generate_graphs.py]: I think this needs to be changed to `reserve_requirements_array` to fix the NameError

**Comment by rconcep** [comment]
Looks good.

**Pull request merged**